### PR TITLE
[vs-code][3/n] Welcome Page: Use Markdown instead of HTML

### PR DESCRIPTION
**Pull request by rossdanlm**
[vs-code][3/n] Welcome Page: Use Markdown instead of HTML

Got Markdown to work. Turns out when I changed the function from normal --> `async`, hot reload no longer worked when doing `yarn && yarn compile` so I had to close the debug vscode window and re-open

## Test Plan

Screenshot 2024-02-08 at 16 38 10

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/1194).
* __->__ #1194
* #1182
* #1178

### CODE REVIEW

**Review by saqadri — APPROVED** [review]

File: vscode-extension/src/welcomePage.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+AIConfig Editor is a tool that allows you to **create, edit, and manage AIConfig files**. This acts as an open-source playground that supports local and remote models from OpenAI, Anthropic, Meta, Google, HuggingFace, Anyscale, and more.  
+**Tip:** To learn more about AIConfigs and why they're useful, check out the FAQ section at the bottom of this page.
+
+![Playing around with the editor](https://s3.amazonaws.com/lastmileai.aiconfig.public/playing_around_with_editor.png)
+
+## Commands
+
+Get started by trying out some of these commands! Press ("CMD" + "SHIFT" + P) to open the command palette and type in "AIConfig":
+
+- `AIConfig: Create New` -> Creates a new file (in JSON or YAML format)
```
Comment (saqadri) [inline vscode-extension/src/welcomePage.md:12]: Now you can change this to 
```
[`AIConfig: Create New`](command:vscode-aiconfig.createAIConfigJSON) 
```

and so forth.

**Review by saqadri — APPROVED** [review]

**Comment by rossdanlm** [comment]
Wasn't able to get the VSCode command to execute, going to land and we can fix forward. 

cc @saqadri, @rholinshead  Can one of you take a look and see if there's anything I missed? Thanks!

https://github.com/lastmile-ai/aiconfig/assets/151060367/b4a1645e-9284-494f-a1f7-9badd8d0688a

**Comment by rholinshead** [comment]
> Wasn't able to get the VSCode command to execute, going to land and we can fix forward.
> 
> cc @saqadri, @rholinshead Can one of you take a look and see if there's anything I missed? Thanks!
> 
>  4bfe1d3b-f5c1-4827-93b7-322cd766db61.mp4

Looking into it. I strongly think it's not supported in markdown preview -- https://github.com/microsoft/vscode/issues/102532

Will look into workarounds

**Comment by rholinshead** [comment]
> Wasn't able to get the VSCode command to execute, going to land and we can fix forward.
> 
> cc @saqadri, @rholinshead Can one of you take a look and see if there's anything I missed? Thanks!
> 
>  4bfe1d3b-f5c1-4827-93b7-322cd766db61.mp4

Ok, I spent some time looking into this today. Command links aren't supported in the markdown preview:
- https://github.com/microsoft/vscode/issues/102532
- suggestion of using markdown.previewScripts doesn't work because you can't access vscode from the script (https://github.com/microsoft/vscode/issues/122961) --> https://github.com/lastmile-ai/aiconfig/pull/1214

I tried a few other things as well, like injecting markdown into html with the markdown="1" flag, using MarkdownString with trusted=true set (note, can't just inject into a webview or MarkdownPreview unfortunately).

Some things to consider:
- There is a "walkthrough" contribution which we could leverage for showing commands, but it's super restrictive; can only show some text / media and command button: https://github.com/microsoft/vscode/issues/83451 
- Could go back to doing HTML, but that makes is harder for non-eng to update
- Could do some (hacky) piecing together of html, using a library to convert our md file to HTML (e.g. showdown) and inject clickable command links into it

IMO not worth the effort to just add links to the commands that people can manually do in the command palette. Tons of other extensions just surface the command palette commands in their extension README page

**Pull request merged**